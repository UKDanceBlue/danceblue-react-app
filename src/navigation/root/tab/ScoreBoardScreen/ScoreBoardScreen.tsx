// import firebaseFirestore from "@react-native-firebase/firestore";
// import { useCallback, useState } from "react";
// import { RefreshControl, SafeAreaView, ScrollView } from "react-native";

// import { useAppSelector } from "../../../../common/CustomHooks";
// import Standings from "../../../../common/components/Standings";
// import { FirestoreTeam } from "../../../../common/firestore/teams";
// import { showMessage } from "../../../../common/util/AlertUtils";
// import { FirestoreMoraleTeam } from "../../../../types/FirebaseTypes";
// import { StandingType } from "../../../../types/StandingType";

/**
 * Wrapper for a Standings component
 */
const ScoreBoardScreen = () => {
  return null;
  // const { pointType } = useAppSelector((state) => state.appConfig.scoreboardMode);
  // const userTeamName = useAppSelector((state) => state.userData.team?.name);
  // // const moraleTeamName = useAppSelector((state) => state);
  // const [ standingData, setStandingData ] = useState<StandingType[]>([]);
  // const [ loading, setLoading ] = useState(true);

  // const refresh = useCallback(() => {
  //   switch (pointType) {
  //   case "spirit":
  //     firebaseFirestore().collection("teams").get()
  //       .then((querySnapshot) => {
  //         const tempStandingData: StandingType[] = [];
  //         querySnapshot.forEach((document) => {
  //           const teamData = document.data() as FirestoreTeam;
  //           // Ensure we don't show the test team on the scoreboard
  //           if (teamData.spiritSpreadsheetId && document.id !== "jR29Y3wJ59evnRaWWKC4") {
  //             tempStandingData.push({
  //               id: document.id,
  //               name: teamData.name,
  //               points: teamData.totalSpiritPoints ?? 0,
  //               highlighted: userTeamName === teamData.name,
  //             });
  //           }
  //         });
  //         setStandingData(tempStandingData);
  //       }).catch(universalCatch);
  //     break;

  //   case "morale":
  //     firebaseFirestore().collection("marathon/2022/morale-teams").get()
  //       .then(
  //         (querySnapshot) => {
  //           const tempStandingData: StandingType[] = [];
  //           querySnapshot.forEach((document) => {
  //             const teamData = document.data() as FirestoreMoraleTeam;
  //             tempStandingData.push({
  //               id: document.id,
  //               name: `Team #${teamData.teamNumber}:\n${teamData.leaders}`,
  //               points: teamData.points,
  //               highlighted: false // moraleTeamId === teamData.teamNumber,
  //             });
  //           });
  //           setStandingData(tempStandingData);
  //         }
  //       ).catch(universalCatch);
  //     break;

  //   case undefined:
  //     setStandingData([]);
  //     break;

  //   default:
  //     showMessage("Failed to load valid point type configuration");
  //     break;
  //   }
  //   return () => {
  //     setLoading(false);
  //   };
  // }, [ pointType, userTeamName ]);

  // return (
  //   <ScrollView
  //     showsVerticalScrollIndicator
  //     refreshControl={<RefreshControl refreshing={loading} onRefresh={refresh} />}
  //   >
  //     <SafeAreaView>
  //       <Standings titleText="Spirit Point Standings" standingData={standingData} startExpanded />
  //     </SafeAreaView>
  //   </ScrollView>
  // );
};

export default ScoreBoardScreen;
