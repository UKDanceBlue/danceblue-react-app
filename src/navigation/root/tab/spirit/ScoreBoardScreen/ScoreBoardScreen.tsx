import { FontAwesome5 } from "@expo/vector-icons";
import firebaseFirestore from "@react-native-firebase/firestore";
import { useNavigation } from "@react-navigation/native";
import { SpiritTeamsRootDoc } from "@ukdanceblue/db-app-common";
import { Pressable, VStack } from "native-base/src/components/primitives";
import { useCallback, useEffect, useState } from "react";
import { RefreshControl, SafeAreaView, ScrollView } from "react-native";

import Jumbotron from "../../../../../common/components/Jumbotron";
import JumbotronTeam from "../../../../../common/components/JumbotronTeam";
import { universalCatch } from "../../../../../common/logging";
import { useAuthData, useUserData } from "../../../../../context";
import { StandingType } from "../../../../../types/StandingType";
import { SpiritStackScreenProps } from "../../../../../types/navigationTypes";

import Scoreboard from "./Scoreboard/Scoreboard";

/**
 * Wrapper for a Standings component
 */
const ScoreBoardScreen = () => {
  const {
    teamId: userTeamId, team
  } = useUserData();
  const dbRole = useAuthData().authClaims?.dbRole ?? "public";
  // const moraleTeamName = useAppSelector((state) => state);
  const [ standingData, setStandingData ] = useState<StandingType[]>([]);
  const [ loading, setLoading ] = useState(true);
  const { navigate } = useNavigation<SpiritStackScreenProps<"Scoreboard">["navigation"]>();

  const refresh = useCallback(() => {
    setLoading(true);
    // switch (pointType) {
    // case "spirit":
    firebaseFirestore().doc("spirit/teams").get()
      .then((querySnapshot) => {
        const rootTeamDataJson = querySnapshot.data() as unknown;
        if (!SpiritTeamsRootDoc.isSpiritTeamsRootDocJson(rootTeamDataJson)) {
          throw new Error("Invalid data type");
        } else {
          const rootTeamData = SpiritTeamsRootDoc.fromJson(rootTeamDataJson);
          setStandingData(Object.entries(rootTeamData.basicInfo).filter(([ ,{ teamClass } ]) => teamClass === "committee" ? dbRole === "committee" : true).map(([ teamId, teamData ]) => {
            return {
              id: teamId,
              highlighted: teamId === userTeamId,
              name: teamData.name,
              points: teamData.totalPoints ?? 0
            };
          }));
        }
      })
      .catch(universalCatch)
      .finally(() => setLoading(false));
    // break;

    // case "morale":
    //   firebaseFirestore().collection("marathon/2022/morale-teams").get()
    //     .then(
    //       (querySnapshot) => {
    //         const tempStandingData: StandingType[] = [];
    //         querySnapshot.forEach((document) => {
    //           const teamData = document.data() as FirestoreMoraleTeam;
    //           tempStandingData.push({
    //             id: document.id,
    //             name: `Team #${teamData.teamNumber}:\n${teamData.leaders}`,
    //             points: teamData.points,
    //             highlighted: false // moraleTeamId === teamData.teamNumber,
    //           });
    //         });
    //         setStandingData(tempStandingData);
    //       }
    //     )
    //     .catch(universalCatch);
    //   break;

    // case undefined:
    //   setStandingData([]);
    //   break;

    // default:
    //   showMessage("Failed to load valid point type configuration");
    //   break;
    // }
  }, [ dbRole, userTeamId ]);

  useEffect(() => {
    refresh();
  }, [refresh]);

  return (
    <ScrollView
      showsVerticalScrollIndicator
      refreshControl={<RefreshControl refreshing={loading} onRefresh={refresh} />}
    >
      <SafeAreaView>
        <VStack>
          {
            team == null
              ? (<Jumbotron
                title="You are not part of a team"
                text=""
                subtitle="If you believe this is an error and you have submitted your spirit points, please contact your team captain or the DanceBlue committee."
                icon="users"
                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
                iconType={FontAwesome5}
                iconColor="blue.500"
              />)
              : (
                <Pressable
                  onPress={() => {
                    navigate("MyTeam");
                  }}
                  _pressed={{ opacity: 0.5 }}
                >
                  <JumbotronTeam team={team.name} />
                </Pressable>
              )
          }
          <Scoreboard title="Spirit Points" data={standingData}/>
        </VStack>
      </SafeAreaView>
    </ScrollView>
  );
};

export default ScoreBoardScreen;
