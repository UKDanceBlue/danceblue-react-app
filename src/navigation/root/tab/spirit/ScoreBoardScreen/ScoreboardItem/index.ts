import ScoreboardItem from "./ScoreboardItem";

export default ScoreboardItem;
