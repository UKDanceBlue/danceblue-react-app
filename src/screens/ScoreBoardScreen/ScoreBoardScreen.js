// Import third-party dependencies
import React from "react";
import { SafeAreaView, ScrollView } from "react-native";

// Import first-party dependencies
import Standings from "./Standings";

/**
 * Wrapper for a Standings component
 * @param {Object} props Properties of the component: navigation, firebase
 * @author Tag Howard
 * @since 1.0.2
 * @class
 */
export class ScoreboardScreen extends React.Component {
    render() {
        return (
        <ScrollView showsVerticalScrollIndicator>
            <SafeAreaView style={{ flex: 1 }}>
            <Standings isExpanded />
            </SafeAreaView>
        </ScrollView>
        );
    }
}

export default ScoreboardScreen