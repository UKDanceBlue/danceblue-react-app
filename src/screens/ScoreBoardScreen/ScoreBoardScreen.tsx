/// <reference types="react" />
import React, { useEffect, useState } from 'react';
import { ActivityIndicator, SafeAreaView, ScrollView } from 'react-native';
import { getDocs, collection } from 'firebase/firestore';
import { useAppSelector } from '../../common/CustomHooks';
import Standings from '../../common/components/Standings';
import { firebaseFirestore } from '../../common/FirebaseApp';
import { globalColors, globalStyles } from '../../theme';
import { StandingType } from '../../types/StandingType';

/**
 * Wrapper for a Standings component
 */
const ScoreBoardScreen = () => {
  const userTeamId = useAppSelector((state) => state.auth.teamId);
  const [standingData, setStandingData] = useState<StandingType[]>([]);

  useEffect(() => {
    let shouldUpdateState = true;
    getDocs(collection(firebaseFirestore, 'teams')).then((querySnapshot) => {
      const tempStandingData: StandingType[] = [];
      querySnapshot.forEach((document) => {
        const teamData = document.data();
        // Ensure we don't show the test team on the scoreboard
        if (teamData.spiritSpreadsheetId && document.id !== 'jR29Y3wJ59evnRaWWKC4') {
          tempStandingData.push({
            id: document.id,
            name: teamData.name,
            points: teamData.totalSpiritPoints,
            highlighted: userTeamId === document.id,
          });
        }
      });
      if (shouldUpdateState) {
        setStandingData(tempStandingData);
      }
    });
    return () => {
      shouldUpdateState = false;
    };
  }, [userTeamId]);

  return (
    <ScrollView showsVerticalScrollIndicator>
      <SafeAreaView style={globalStyles.genericView}>
        {standingData.length === 0 && (
          <ActivityIndicator
            size="large"
            color={globalColors.lightBlue}
            style={{
              alignItems: 'center',
              justifyContent: 'center',
              padding: 20,
            }}
          />
        )}
        <Standings titleText="Spirit Point Standings" standingData={standingData} startExpanded />
      </SafeAreaView>
    </ScrollView>
  );
};

export default ScoreBoardScreen;
