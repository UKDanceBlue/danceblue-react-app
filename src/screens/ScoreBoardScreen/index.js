// Allow importing from the parent folder rather than individual files
// for example you can do:   *import { EventScreen } from '../EventScreen'*
// instead of:               *import { EventScreen } from '../EventScreen/EventScreen'*

import Place from './Place';
import Standings from './Standings';
import ScoreboardScreen from './ScoreBoardScreen';

export { Place, Standings };
export default ScoreboardScreen;
