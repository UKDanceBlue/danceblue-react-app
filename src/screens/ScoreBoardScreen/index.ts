// Allow importing from the parent folder rather than individual files
// for example you can do:   *import { EventScreen } from '../EventScreen'*
// instead of:               *import { EventScreen } from '../EventScreen/EventScreen'*

import ScoreboardScreen from "./ScoreBoardScreen";

export default ScoreboardScreen;
