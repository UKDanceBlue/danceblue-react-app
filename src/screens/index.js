import React from 'react'
import { StyleSheet, Dimensions, Image } from 'react-native'
import { createStackNavigator } from '@react-navigation/stack'
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
import Icon from 'react-native-vector-icons/FontAwesome5'
import { BlurView } from 'expo-blur'

import { withFirebaseHOC } from '../../config/Firebase'

import HomeScreen from './Home'
import {ScoreboardScreen } from './Home'
import EventsScreen from './Events'
import { StoreScreen } from './Store'
import MoreScreen from './More'
import Event from '../components/event/Event'

// import ProfileScreen from './More/Profile'
import { FAQScreen } from './More/FAQ'
import { DonateScreen } from './More/Donate'
import { AboutScreen } from './More/About'
import ProfileScreen from './More/Profile'

import { SplashLoginScreen } from './Modals'

const RootStack = createStackNavigator()
const MainStack = createStackNavigator()
const Tabs = createBottomTabNavigator()

function TabsScreen () {
  return (
    <Tabs.Navigator
      screenOptions={
        ({ route }) => ({
        tabBarIcon: ({ color, size }) => {
          const iconMap = new Map([ //Key: Screen   Value: Icon ID
            ['Home'], ['home'],
            ['Events'], ['calendar'],
            ['Store'], ['store'],
            ['More'], ['ellipsis-h'],
            ['Test'], ['ellipsis-h']
          ]);

          // You can return any component that you like here!
          return (
            <Icon
              name={iconMap.get(route.name)}
              size={size}
              color={color}
              style={{ textAlignVertical: 'center' }}
            />
          )
        }
      })}
      tabBarOptions={{
        activeTintColor: 'white',
        activeBackgroundColor: '#3248a8'
      }}
    >
      <Tabs.Screen name='Home' component={HomeScreen} />
      <Tabs.Screen name='Events' component={EventsScreen} />
      <Tabs.Screen name='Store' component={StoreScreen} />
      <Tabs.Screen name='More' component={MoreScreen} />
    </Tabs.Navigator>
  )
}

function MainStackScreen () {
  return (
    <MainStack.Navigator>
      <MainStack.Screen name='Tab' component={TabsScreen} options={{ headerShown: false }} />
      <MainStack.Screen name='Profile' component={ProfileScreen} />
      <MainStack.Screen name='FAQ' component={FAQScreen} />
      <MainStack.Screen name='Donate' component={DonateScreen} />
      <MainStack.Screen name='About' component={AboutScreen} />
      <MainStack.Screen name='Scoreboard' component={ScoreboardScreen} />
      <MainStack.Screen
        name='Event' component={Event} options={({ route }) => ({
          title: route.params.name,
          headerTransparent: true,
          headerMode: 'screen',
          headerBackground: () => (
            <BlurView tint='light' intensity={100} style={StyleSheet.absoluteFill} />
          )
        })}
      />
    </MainStack.Navigator>
  )
}

class RootStackScreen extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      userID: null,
      isLoading: true
    }
  }

  componentDidMount() {
    this.props.firebase.checkAuthUser(user => {
      if (user !== null) this.setState({ userID: user.uid, isLoading: false })
      else this.setState({ isLoading: false })
    })
  }

  render() {
    if (this.state.isLoading) return <Image style={{ width: Dimensions.get('window').width, height: Dimensions.get('window').height }} source={require('../../assets/splash2.png')} />
    return (
      <RootStack.Navigator mode='modal'>
        {this.state.userID && (
          <RootStack.Screen name='Main' component={MainStackScreen} options={{ headerShown: false }} />
        )}
        {!this.state.userID && (
          <RootStack.Screen name='SplashLogin' component={SplashLoginScreen} options={{ headerShown: false }} />
        )}
      </RootStack.Navigator>
    )
  }
}

export default withFirebaseHOC(RootStackScreen)
