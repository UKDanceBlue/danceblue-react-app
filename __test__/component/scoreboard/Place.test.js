import React from 'react';
import renderer from 'react-test-renderer';
import { loadAsync } from 'expo-font';
import { FontAwesome5 } from '@expo/vector-icons';
import { Place } from '../../../src/screens/ScoreBoardScreen';

beforeAll(async () => loadAsync(FontAwesome5.font));

describe('Place component', () => {
  test('renders correctly normally', () => {
    const tree = renderer
      .create(<Place rank={4} teamName="Team Name" teamNumber={69} points={420} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly highlighted', () => {
    const tree = renderer
      .create(<Place isHighlighted rank={4} teamName="Team Name" teamNumber={69} points={420} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly 1st place', () => {
    const tree = renderer
      .create(<Place rank={1} teamName="Team Name" teamNumber={69} points={420} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly 2nd place', () => {
    const tree = renderer
      .create(<Place rank={1} teamName="Team Name" teamNumber={69} points={420} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly 3rd place', () => {
    const tree = renderer
      .create(<Place rank={1} teamName="Team Name" teamNumber={69} points={420} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
