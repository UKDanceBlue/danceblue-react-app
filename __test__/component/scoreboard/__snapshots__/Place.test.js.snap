// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Place component renders correctly 1st place 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#1F2236",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "20%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 20,
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
    >
      1
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "gold",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "700",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "textAlign": "center",
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 15,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "textAlign": "right",
        "width": "30%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 17,
            "paddingRight": 10,
          },
        ]
      }
    >
      420
       points
    </Text>
  </View>
</View>
`;

exports[`Place component renders correctly 2nd place 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#1F2236",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "20%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 20,
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
    >
      1
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "gold",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "700",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "textAlign": "center",
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 15,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "textAlign": "right",
        "width": "30%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 17,
            "paddingRight": 10,
          },
        ]
      }
    >
      420
       points
    </Text>
  </View>
</View>
`;

exports[`Place component renders correctly 3rd place 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#1F2236",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "20%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 20,
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
    >
      1
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "gold",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "700",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "textAlign": "center",
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 15,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "textAlign": "right",
        "width": "30%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 17,
            "paddingRight": 10,
          },
        ]
      }
    >
      420
       points
    </Text>
  </View>
</View>
`;

exports[`Place component renders correctly highlighted 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "#1F2236",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 10,
        "paddingTop": 10,
      },
      Object {
        "backgroundColor": "#8BA9FC",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "20%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 20,
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
    >
      4
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "textAlign": "center",
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 15,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "textAlign": "right",
        "width": "30%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 17,
            "paddingRight": 10,
          },
        ]
      }
    >
      420
       points
    </Text>
  </View>
</View>
`;

exports[`Place component renders correctly normally 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#1F2236",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "20%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 20,
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
    >
      4
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "textAlign": "center",
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 15,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "textAlign": "right",
        "width": "30%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 17,
            "paddingRight": 10,
          },
        ]
      }
    >
      420
       points
    </Text>
  </View>
</View>
`;
